### PR TITLE
Select all within a folder for upload

### PR DESCRIPTION
**Pull request by AndyScherzinger**
see $subject reimplementation of an oC PR with state handling of the select icon in the action bar.

This PR add the ability to do a select/unselect all when uploading files from within the NC app.

This PR has been requested in the Play store review and also by users for ownCloud sometime last year

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
Please review @przybylski @tobiasKaminsky

File: src/com/owncloud/android/ui/fragment/LocalFileListFragment.java [hunk 1]
```diff
@@ -113,6 +113,7 @@ public void onItemClick(AdapterView<?> l, View v, int position, long id) {
                 listDirectory(file);
                 // notify the click to container Activity
                 mContainerActivity.onDirectoryClick(file);
+                Log_OC.e(TAG, "onItemClick directory");
```
Comment (przybylski) [inline src/com/owncloud/android/ui/fragment/LocalFileListFragment.java]: remove please :)

**Comment by przybylski** [comment]
👍

**Comment by przybylski** [comment]
:shipit:

**Comment by przybylski** [comment]
odd, my LGTM is not accepted

**Comment by AndyScherzinger** [comment]
Na, it should be but we defined that it needs two approvals, so @tobiasKaminsky also need to approve it or @LukasReschke or @jancborchardt 
I can't since the PR creator can't approve which makes sense ;)

**Comment by przybylski** [comment]
It might be a problem if one of us is away ;)

**Comment by AndyScherzinger** [comment]
Na, then you can still ping anybody from the list :)

File: src/com/owncloud/android/ui/fragment/LocalFileListFragment.java (hunk 1 above)
Comment (AndyScherzinger) [inline src/com/owncloud/android/ui/fragment/LocalFileListFragment.java]: DONE, see https://github.com/nextcloud/android/pull/49/commits/9c677a47aa6716b60306c66469a0c52214e8517a

**Comment by AndyScherzinger** [comment]
@ all added another commit https://github.com/nextcloud/android/pull/49/commits/fb9cf68aa65d9f6ccb77a8cfc8bc375c0997ea3c which fixes a code duplicate I missed.

**Comment by przybylski** [comment]
👍

**Comment by tobiasKaminsky** [comment]
:+1:

**Pull request merged**